Task-58927 : Fix copied link of shared document does not work

sofyenne opened the pull request:
ISSUE : When users copy the link of the shared document and past it to preview this document , the document not opened and users redirect to the folder view of application .
FIX :  add `sourceId` of shared document to copied link

halaya committed:
TASK-35704: Create FB maintenance and update projects versions/dependencies

halaya committed:
TASK-35704: Create FB maintenance and update projects versions/dependencies

halaya committed:
TASK-35704: Create FB maintenance and update projects versions/dependencies

halaya committed:
TASK-35704: Create FB maintenance and update projects versions/dependencies

halaya committed:
TASK-35704: Create FB maintenance and update projects versions/dependencies

azayati committed:
Task-58612: [Accessibility - Contrast C3.2 - Stream Page] Fix contrast problem for all "little" texts (#415)

azayati committed:
Task-58612: [Accessibility - Contrast C3.2 - Stream Page] Fix contrast problem for all "little" texts (#415)

azayati committed:
Task-58612: [Accessibility - Contrast C3.2 - Stream Page] Fix contrast problem for all "little" texts (#415)

azayati committed:
Task-58612: [Accessibility - Contrast C3.2 - Stream Page] Fix contrast problem for all "little" texts (#415)

hchaabene committed:
TASK-57797: Hide system folders and files (#421)

Prior to this change, the system folders and files displayed inside the personal drives such as Folksonomy Searches exo:thumbnails
This change these system files are hiding

ahamdi committed:
Task-58606: can't add files inside a folder which name contains special characters (#426)

ahamdi committed:
Task-58606: can't add files inside a folder which name contains special characters (#426)

sofyenne committed:
Task-58927 : Fix copied link of chered document does not work

ahamdi reviewed (changes requested):
The preview should work for the symlink , when we get the ID of the symlink, we have to load the original content and preview it correctly

sofyenne committed:
Task-58997: Add the ability to create folder with the name contains only numbers

ISSUE : On new document application users can't create a folder with the name contains only numbers
FIX : Remove the condition who verify if the name of folder is contain only numbers and can't able user to create this folder

sofyenne committed:
Task-58708: Fix redirect to root folder when duplicate file 

ISSUE : When duplicate document inside any folder in any level the document duplicated but the current location changed to the root folder of document app
FIX : fix the `getFolderPath` method to get correctly the current folder path where the file duplicate

sofyenne committed:
Task-58708: Fix redirect to root folder when duplicate file 

ISSUE : When duplicate document inside any folder in any level the document duplicated but the current location changed to the root folder of document app
FIX : fix the `getFolderPath` method to get correctly the current folder path where the file duplicate

sofyenne committed:
Merge branch 'feature/maintenance' into Task-58927

wassimmhamdi committed:
Task-58900: In move drawer when changing the destination folder the current location shouldn't be changed (#432)

Problem: when click on the the move button of a folder in documents application of a spaceX, the hierarchy icon of the Current Location and destination in the opened move drawer are clickable
Fix: we add a property to assure that these two tree views are hidden and not clickable

wassimmhamdi committed:
Task-58900: In move drawer when changing the destination folder the current location shouldn't be changed (#432)

Problem: when click on the the move button of a folder in documents application of a spaceX, the hierarchy icon of the Current Location and destination in the opened move drawer are clickable
Fix: we add a property to assure that these two tree views are hidden and not clickable

sofyenne committed:
update open preview method

sofyenne commented on documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileNameCell.vue:
```diff
@@ -243,10 +243,7 @@ export default {
-        if (this.file.sourceID){
-          id = this.file.sourceID;
-        }
+        let id = this.file.sourceID ? this.file.sourceID :this.file.id;
```
@ahamdi  the `sourceId` is the id of original file so the condition to check if the file is a symlink is already exist ( if sourceId not null so the file is a symlink ) and get the original attachment by the id or the sourceId  that's why the open preview document works correctly , i just clean the code .

ahamdi replied:
I agree, and think this is how it was implemented. But the behavior is not correct, we need to be able to preview a symlink the same way we preview original document.

sofyenne commented on documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileNameCell.vue:
```diff
@@ -271,7 +268,7 @@ export default {
           .catch(e => console.error(e))
           .finally(() => {
-            window.history.pushState('', '', `${eXo.env.server.portalBaseURL}?documentPreviewId=${this.file.id}`);
+            window.history.pushState('', '', `${eXo.env.server.portalBaseURL}?documentPreviewId=${this.file.sourceID ? this.file.sourceID : this.file.id}`);
```
@ahamdi  i add the condition to check if the file is a symlink , to get the original attachment when user reload the page

ahamdi committed:
Merge branch 'feature/maintenance' into Task-58927

sofyenne commented on documents-webapp/src/main/webapp/vue-app/documents/components/body/actions/CopyLinkMenuAction.vue:
```diff
@@ -32,7 +32,7 @@ export default {
         path = `${path}?folderId=${this.file.id}`;
       } else {
-        path = `${path}?documentPreviewId=${this.file.id}`;
+        path = `${path}?documentPreviewId=${this.file.sourceID ? this.file.sourceID:this.file.id}`;
```
@ahamdi i add the same condition to the copied link to get the original file and that fix the problem .WDYT ?

ahamdi committed:
Merge branch 'feature/maintenance' into Task-58927

ahamdi committed:
Merge branch 'feature/maintenance' into Task-58927
